src/test_bst.py

```diff
@@ -186,14 +208,94 @@ def test_in_order():
     assert next(gen) is 6
     assert next(gen) is 7
 
-def test_post_order():
-    """test pre-order first"""
-    new_tree = BinarySearchTree([4,2,3,1,6,5,7])
-    gen = new_tree.post_order()
+
+def test_post_order(tree_init_list):
+    """Test pre-order first."""
+    gen = tree_init_list.post_order()
     assert next(gen) is 1
```

Comment:
Multiple assert lines are sort of controversial in the community. Your test should be written to test one thing so should  have one assert line, I don't mind if they are closely related asserts.

To cut down your lines of code, I would build a list using your traversal method and asserting if the resulting list matches and expected list. For example:

```
lst = [n for n in tree_init_list.post_order()]
assert lst == [1,3,2,5,7,6,4]
```